pkg/rhcos/builds.go

```diff
@@ -23,7 +25,8 @@ type metadata struct {
 	OSTreeVersion string `json:"ostree-version"`
 }
 
-func fetchRHCOSBuild(ctx context.Context) (*metadata, error) {
+// FetchRHCOSBuild retrieves the pinned RHEL CoreOS version.
+func FetchRHCOSBuild(ctx context.Context) (*Metadata, error) {
```

Comment:
Also the use-case in this PR does not justify the reason to make this function public, as it opens the package a really big API surface.